src/NServiceBus.Testing.Fakes/Fakes/TestableInvokeHandlerContext.cs

```diff
@@ -0,0 +1,83 @@
+﻿namespace NServiceBus.Testing
+{
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using NServiceBus.Persistence;
+    using NServiceBus.Pipeline;
+    using NServiceBus.Unicast.Messages;
+
+    /// <summary>
+    /// A testable implementation of <see cref="IInvokeHandlerContext" />.
+    /// </summary>
+    public class TestableInvokeHandlerContext : TestableIncomingContext, IInvokeHandlerContext
+    {
+        /// <summary>
+        /// Creates a new instance of <see cref="TestableInvokeHandlerContext" />.
+        /// </summary>
+        public TestableInvokeHandlerContext(IMessageCreator messageCreator = null) : base(messageCreator)
+        {
+        }
+
+        /// <summary>
+        /// Indicates if <see cref="IMessageHandlerContext.DoNotContinueDispatchingCurrentMessageToHandlers" /> has been called.
+        /// </summary>
+        public bool DoNotContinueDispatchingCurrentMessageToHandlersWasCalled { get; set; }
+
+        /// <summary>
+        /// Moves the message being handled to the back of the list of available
+        /// messages so it can be handled later.
+        /// </summary>
+        public virtual Task HandleCurrentMessageLater()
+        {
+            HandleCurrentMessageLaterWasCalled = true;
+            return Task.FromResult(0);
+        }
+
+        /// <summary>
+        /// Indicates if <see cref="IMessageHandlerContext.HandleCurrentMessageLater" /> has been called.
+        /// </summary>
+        public bool HandleCurrentMessageLaterWasCalled { get; set; }
```

Comment:
Fair point, but in this case, when would you use the setter and not the method? (what's the use case?)